src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,164 @@
+
+    /**
+     * Constructs a TaskList object with an existing ArrayList.
+     * @param list The ArrayList of tasks.
+     */
+    public TaskList(ArrayList<Task> list) {
+        this.tasks = list;
+    }
+
+    /**
+     * Constructs a TaskList object with a new ArrayList.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     * @return The integer value of the number of tasks in the list.
+     */
+    public int length() {
+        return tasks.size();
+    }
+
+    /**
+     * Gets the Task at the specific index.
+     * @param index The task's index.
+     * @return The Task at the index.
+     */
+    public Task get(int index) {
+        return tasks.get(index);
+    }
+
+
+    /**
+     * Marks the Task in the ArrayList as completed.
+     * @param input The user input.
+     * @param storage The Storage object.
+     * @param ui The Ui object.
+     * @throws DukeException Thrown in the event the Task cannot be accessed.
+     */
+    public String handleDone(String input, Storage storage, Ui ui) throws DukeException {
+        try {
+            int taskNumber = Integer.parseInt(input.substring(5));
+            Task currTask = tasks.get(taskNumber - 1);
+            currTask.markAsDone();
+            storage.saveTask(tasks);
+            return ui.done(currTask);
+        } catch (StringIndexOutOfBoundsException err) {
+            throw new DukeException("can type properly? liddis: 'done taskNumber'");
+        } catch (NumberFormatException err) {
+            throw new DukeException("can enter number only pls? dun anyhow");
+        } catch (ArrayIndexOutOfBoundsException err) {
+            throw new DukeException("you where got so many tasks?");
+        }
+    }
+
+    private String add(Task task, Storage storage, Ui ui) {
+        tasks.add(task);
+        storage.saveTask(tasks);
+        return ui.add(task, tasks.size());
+    }
+
+    /**
+     * Adds a Todo object into the ArrayList.
+     * @param input The user input.
+     * @param storage The Storage object.
+     * @param ui The Ui object.
+     * @throws DukeException Thrown in the event the user input is wrong.
+     */
+    public String handleTodo(String input, Storage storage, Ui ui) throws DukeException {
+        try {
+            Todo todo = new Todo(input.substring(5));
+            return add(todo, storage, ui);
+        } catch (StringIndexOutOfBoundsException err) {
+            throw new DukeException("u never say what to do?");
+        }
+    }
+
+
+    /**
+     * Adds a Deadline object into the ArrayList.
+     * @param input The user input.
+     * @param storage The Storage object.
+     * @param ui The Ui object.
+     * @throws DukeException Thrown in the event the user input is wrong.
+     */
+    public String handleDeadline(String input, Storage storage, Ui ui) throws DukeException {
+        try {
+            int split = input.indexOf("/");
```

Comment:
Perhaps there could be a more specific name for the variable `split`, like `splitAtIdx`?